waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -54,45 +64,65 @@ proc insertOrReplace(ps: PeerStorage,
 proc dialPeer(pm: PeerManager, peerId: PeerID,
               addrs: seq[MultiAddress], proto: string,
               dialTimeout = DefaultDialTimeout,
-              source = "api"
+              source = "api",
+              respectBackoff = false,
               ): Future[Option[Connection]] {.async.} =
 
   # Do not attempt to dial self
   if peerId == pm.switch.peerInfo.peerId:
     return none(Connection)
 
-  info "Dialing peer from manager", wireAddr = addrs, peerId = peerId
+  let failedAttempts = pm.peerStore[NumberFailedConnBook][peerId]
+
+  # Respect a backoff period for peers that have failed to connect
+  if respectBackoff and not pm.peerStore.canBeConnected(peerId, pm.initialBackoffInSec):
+    debug "Peer recently failed to connect, skipping dial",
+          peerId = peerid,
+          failedAttempts = failedAttempts,
+          lastFailed = pm.peerStore[LastFailedConnBook][peerId]
+    return none(Connection)
```

Comment:
Afaict this will not allow any differentiation between "still backing off" and other connection failures. Would this be something applications require? In other words, this would not/should not stop the application from attempting to dial this peer continuously as there is no indication that under the hood we're still backing off?